lib/ddtrace/transport/http/adapters/net.rb

```diff
@@ -20,8 +20,14 @@ def initialize(hostname, port, options = {})
           end
 
           def open
-            # Open connection
-            ::Net::HTTP.start(hostname, port, open_timeout: timeout, read_timeout: timeout) do |http|
+            # DEV Initializing +Net::HTTP+ directly help us avoid expensive
+            # options processing done in +Net::HTTP.start+:
+            # https://github.com/ruby/ruby/blob/b2d96abb42abbe2e01f010ffc9ac51f0f9a50002/lib/net/http.rb#L614-L618
+            req = ::Net::HTTP.new(hostname, port, nil)
```

Comment:
I know this is kinda out there but have we considered using a 3rd party http library? Might not be worth the pain but I believe some other vendors use [`http.rb`](https://github.com/honeycombio/libhoney-rb/blob/00dd785ce796c193db0a628fd4be251a2353fe69/libhoney.gemspec#L37)

Reply:
We have a follow up task to investigate this 👍